From f8c36d8eaa1516721756e50ed8c695b46a11a288 Mon Sep 17 00:00:00 2001
From: Bin Huang <binhuang@meta.com>
Date: Tue, 11 Apr 2023 14:14:00 +0700
Subject: [PATCH] ARM: dts: aspeed: Add Facebook montblanc BMC

Add initial version of device tree for Facebook montblanc (AST2600) BMC.

montblanc is Facebook's next generation switch platform with an AST2600
BMC integrated for health monitoring purpose.

Signed-off-by: Bin Huang <binhuang@meta.com>
---
 .../dts/aspeed-bmc-facebook-montblanc.dts     | 36 +++++++++++++++++++
 1 file changed, 36 insertions(+)
 create mode 100644 arch/arm/boot/dts/aspeed-bmc-facebook-montblanc.dts

diff --git a/arch/arm/boot/dts/aspeed-bmc-facebook-montblanc.dts b/arch/arm/boot/dts/aspeed-bmc-facebook-montblanc.dts
new file mode 100644
index 000000000000..a605245695a9
--- /dev/null
+++ b/arch/arm/boot/dts/aspeed-bmc-facebook-montblanc.dts
@@ -0,0 +1,36 @@
+// SPDX-License-Identifier: GPL-2.0+
+// Copyright (c) Meta Platforms, Inc. and affiliates.
+
+/dts-v1/;
+
+#include "ast2600-facebook-netbmc-common.dtsi"
+
+/ {
+	model = "Facebook montblanc BMC";
+	compatible = "facebook,montblanc-bmc", "aspeed,ast2600";
+
+	chosen {
+		stdout-path = &uart1;
+	};
+};
+
+&mdio1 {
+	status = "okay";
+
+	ethphy3: ethernet-phy@0 {
+		compatible = "ethernet-phy-id0141.0dd1",
+					"ethernet-phy-ieee802.3-c22";
+		reg = <0x00>;
+		marvell,reg-init =
+			<0x2  0x15 0xffcf 0x0000>, // disable RGMII Delay
+		    <0x12 0x14 0x7ff8 0x8002>; // mode RGMII to 1000Base-X
+	};
+};
+
+&mac3 {
+	status = "okay";
+	phy-mode = "rgmii";
+	phy-handle = <&ethphy3>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_rgmii4_default>;
+};
--
2.25.1
